c/misra/src/rules/DIR-4-9/FunctionOverFunctionLikeMacro.ql

```diff
@@ -0,0 +1,73 @@
+/**
+ * @id c/misra/function-over-function-like-macro
+ * @name DIR-4-9: A function should be used in preference to a function-like macro where they are interchangeable
+ * @description Using a function-like macro instead of a function can lead to unexpected program
+ *              behaviour.
+ * @kind problem
+ * @precision medium
+ * @problem.severity recommendation
+ * @tags external/misra/id/dir-4-9
+ *       external/misra/audit
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/advisory
+ */
+
+import cpp
+import codingstandards.c.misra
+import codingstandards.cpp.FunctionLikeMacro
+import codingstandards.cpp.Naming
+
+predicate isOperator(string possible) {
+  possible in [
+      "+", "-", "*", "/", "%", "^", "&", "|", "~", "!", "=", "<", ">", "+=", "-=", "*=", "/=", "%=",
+      "^=", "&=", "|=", "<<", ">>", ">>=", "<<=", "==", "!=", "<=", ">=", "<=>", "&&", "||", "++",
+      "--", "->*", "->", "()", "[]"
+    ]
+}
+
+//cases where we trust the choice
+predicate omission(MacroInvocation i) {
+  i.getFile() instanceof HeaderFile or
```

Comment:
Could you help me understand the first omission?

Reply:
its a pretty arbitrary heuristic , the idea behind it is to narrow to code (source files) with more churn I guess (assuming thats src > headers)

Reply:
I have omitted that heuristic now, but I still think there may be too many results as is